diff --git a/dist/modifiers/velcro.js b/dist/modifiers/velcro.js
index f4df71a4fafbb5757e38788d95f9f25285ff35dc..187825e071bb665191ee0b51965e5e66ef6b8fde 100644
--- a/dist/modifiers/velcro.js
+++ b/dist/modifiers/velcro.js
@@ -52,10 +52,13 @@ class VelcroModifier extends Modifier {
         placement,
         strategy
       });
+
+      const { referenceHidden } = middlewareData.hide;
       Object.assign(floatingElement.style, {
         top: `${y}px`,
         left: `${x}px`,
-        margin: 0
+        margin: 0,
+        visibility: referenceHidden ? "hidden" : "visible"
       });
       setVelcroData?.(middlewareData.metadata);
     };